### ISSUE TITLE
'QgsAbstractProcessingParameterWidgetWrapper' object has no attribute 'value'

### ISSUE THREAD

**Issue by gacarrillor** [qgis/QGIS#35516]
On current master (commit https://github.com/qgis/QGIS/commit/acf72732208d811de526c1ef90b50517b2db8ed6)

Steps to reproduce:

1. Open QGIS.
2. Load 2 layers.
3. Open the Refactor Fields dialog.
4. Press the Escape key.
5. Remove the layer that is currently selected in the Layers Panel.
6. You get the following error: 

```
Traceback (most recent call last):
  File "/docs/dev/qgis/QGIS/build_20200401/output/python/plugins/processing/algs/qgis/ui/FieldsMappingPanel.py", line 592, in parentLayerChanged
    self.setLayer(self.sender().value())
AttributeError: 'QgsAbstractProcessingParameterWidgetWrapper' object has no attribute 'value'

```
processing 2.12.99, QGIS 3.13.0-Master on Linux 4.15.0-20-generic

**Comment by nyalldawson** [comment]
Fixed in master

**Comment by gacarrillor** [comment]
This one is fixed by #34796, but the described bug is still there. Since instructions to reproduce changed, I created another issue: https://github.com/qgis/QGIS/issues/36582
